pkg/controllers/controllers.go

```diff
@@ -16,13 +16,16 @@ package controllers
 
 import (
 	"github.com/aws/aws-sdk-go/service/sqs"
+	"knative.dev/pkg/logging"
 
 	"github.com/aws/karpenter-core/pkg/operator/controller"
 	awscontext "github.com/aws/karpenter/pkg/context"
 	"github.com/aws/karpenter/pkg/controllers/interruption"
+	"github.com/aws/karpenter/pkg/utils/project"
 )
 
 func NewControllers(ctx awscontext.Context) []controller.Controller {
+	logging.FromContext(ctx).Infof("Initializing with version %s", project.Version)
```

Comment:
Does it make sense to put this in the main?

Reply:
It does make sense, in the sense that it's better to call it as soon as possible.

However it seems like the logging instrumentation happens in `karpenter-core` as the first few lines of the controller are emitted by it:

```
2022-11-16T21:23:25.057Z	DEBUG	Successfully created the logger.
2022-11-16T21:23:25.057Z	DEBUG	Logging level set to: debug
{"level":"info","ts":1668633805.0638587,"logger":"fallback","caller":"injection/injection.go:61","msg":"Starting informers..."}
2022-11-16T21:23:25.164Z	DEBUG	controller	Waiting for settings ConfigMap(s) creation	{"commit": "a00f464-dirty"}
2022-11-16T21:23:25.664Z	DEBUG	controller	Settings ConfigMap(s) exist	{"commit": "a00f464-dirty"}
2022-11-16T21:23:25.669Z	DEBUG	controller	karpenter-global-settings config "karpenter-global-settings" config was added or updated: settings.Settings{BatchMaxDuration:v1.Duration{Duration:10000000000}, BatchIdleDuration:v1.Duration{Duration:1000000000}}	{"commit": "a00f464-dirty"}
2022-11-16T21:23:25.669Z	DEBUG	controller	karpenter-global-settings config "karpenter-global-settings" config was added or updated: settings.Settings{ClusterName:"rmeast1test1", ClusterEndpoint:"https://9F91372BD5C47CAC6A6FC08B30AD478E.gr7.us-east-1.eks.amazonaws.com", DefaultInstanceProfile:"KarpenterNodeInstanceProfile-rmeast1test1", EnablePodENI:false, EnableENILimitedPodDensity:true, IsolatedVPC:false, NodeNameConvention:"ip-name", VMMemoryOverheadPercent:0.075, InterruptionQueueName:"rmeast1test1", Tags:map[string]string{}}	{"commit": "a00f464-dirty"}
2022-11-16T21:23:25.781Z	DEBUG	controller.aws	Using AWS region us-east-1	{"commit": "a00f464-dirty"}
2022-11-16T21:23:25.786Z	DEBUG	controller.aws	Discovered DNS IP 10.100.0.10	{"commit": "a00f464-dirty"}
2022/11/16 21:23:25 Registering 2 clients
2022/11/16 21:23:25 Registering 2 informer factories
2022/11/16 21:23:25 Registering 3 informers
2022/11/16 21:23:25 Registering 6 controllers
2022-11-16T21:23:25.959Z	DEBUG	controller.aws	Discovered caBundle, length 1066	{"commit": "a00f464-dirty"}
2022-11-16T21:23:25.959Z	INFO	controller.aws	Initializing with version v0.18.0-143-ga00f4643	{"commit": "a00f464-dirty"}
```

That's why I just put this back were it was before, I'm not sure if the logging is ready to be used in main but can look into it